src/sdk/pynni/nni/compression/torch/builtin_pruners.py

```diff
@@ -471,3 +490,239 @@ def _get_distance_sum(self, weight, in_idx, out_idx):
 
     def update_epoch(self, epoch):
         self.mask_calculated_ops = set()
+
+
+class ActivationRankFilterPruner(Pruner):
+    """
+    A structured pruning base class that prunes the filters with the smallest
+    importance criterion in convolution layers to achieve a preset level of network sparsity.
+    Hengyuan Hu, Rui Peng, Yu-Wing Tai and Chi-Keung Tang,
+    "Network Trimming: A Data-Driven Neuron Pruning Approach towards Efficient Deep Architectures", ICLR 2016.
+    https://arxiv.org/abs/1607.03250
+    Pavlo Molchanov, Stephen Tyree, Tero Karras, Timo Aila and Jan Kautz,
+    "Pruning Convolutional Neural Networks for Resource Efficient Inference", ICLR 2017.
+    https://arxiv.org/abs/1611.06440
+    """
```

Comment:
this is also a one-shot pruning algorithm?

Reply:
yes